src/services/Middleware.ts

```diff
@@ -0,0 +1,54 @@
+import * as Hapi from 'hapi'
+import * as Boom from 'boom'
+import { create as createLogger } from '../common/log'
+import Config from '../services/Config'
+import { Injector } from 'reduct'
+const log = createLogger('Middleware')
+
+interface DecoratedToolkit extends Hapi.ResponseToolkit {
+  config?: Config
+}
+
+export default class Middleware {
+  public config: Config
+  private decorations: object = {
+    'config': () => { return this.config }
+  }
+
+  constructor (deps: Injector) {
+    this.config = deps(Config)
+  }
+
+  async checkSelfTest (request: Hapi.Request, h: DecoratedToolkit) {
+    const config = h.config
```

Comment:
If you use `checkSelfTest.bind(this)` instead of using checkSelfTest, you can keep the config in scope. That way you don't need a special decorator to inject the config into the hapi object.